### PR TITLE
fix: refactor service level in pkg/dashboard/apiserver/experiment

### PR DESCRIPTION
**Pull request by ConnieHan2019**
### What problem does this PR [solve?](https://github.com/chaos-mesh/chaos-mesh/issues/2875)
#2875 

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.2
  - [ ] release-2.1

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [x] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: pkg/dashboard/apiserver/experiment/experiment.go [hunk 1]
```diff
@@ -60,12 +60,14 @@ func NewService(
...
 	config *config.ChaosDashboardConfig,
 	scheme *runtime.Scheme,
+	log logr.Logger,
 ) *Service {
 	return &Service{
 		archive: archive,
 		event:   event,
 		config:  config,
 		scheme:  scheme,
+		log: log,
```
Comment (xlgao-zju) [inline pkg/dashboard/apiserver/experiment/experiment.go:70]: use `u.Log.WithName("experiments")` would be better

**Review by xlgao-zju — CHANGES_REQUESTED** [review]
and we can remove the Line 48, since we will not use  the var `log` anymore.

**Comment by xlgao-zju** [comment]
you can run `make check` before commit.

File: pkg/dashboard/apiserver/experiment/experiment.go (hunk 1 above)
Comment (g1eny0ung) [reply pkg/dashboard/apiserver/experiment/experiment.go:71]: This is a legacy method to log something, so this part is not a problem.

**Review by xlgao-zju — APPROVED** [review]
LGTM

**Pull request merged**